### PR TITLE
Add dichotomy algorithm

### PR DESCRIPTION
**Pull request by kofgokhan**
Link for the algorithm: https://doi.org/10.1287/mnsc.25.1.73

Need to add more tests (and the filter for weak dominance I think) and the docstrings. Test case taken from the original paper by Y. P. Aneja, K. P. K. Nair (1979)

### CODE REVIEW

File: src/algorithms/Dichotomy.jl [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        while !isempty(L)
+            r, s = pop!(L)
+            yr, ys = solutions[r].y, solutions[s].y
+            a1 = abs(ys[2] - yr[2])
+            a2 = abs(ys[1] - yr[1])
+            MOI.set(
+                model.inner,
+                MOI.ObjectiveFunction{typeof(f1 + f2)}(),
+                a1 * f1 + a2 * f2,
+            )
```
Comment (odow) [inline src/algorithms/Dichotomy.jl:65]: Hmm. This looks suspiciously similar. Is Dichotomy just the Non-inferior set estimation method by a different name?

https://github.com/jump-dev/MultiObjectiveAlgorithms.jl/blob/b03c3ffceec4e1f71e26a9eda0a02e03f6510cdb/src/algorithms/NISE.jl#L54-L79

Comment (kofgokhan) [reply src/algorithms/Dichotomy.jl:65]: I could not access the paper but they look the same. Return the same thing for the sample problem. Both papers from 1979. One in Jan the other in October. Dichotomy can be better known so maybe just change the name of NISE.jl? I don't know.

Comment (xgandibleux) [reply src/algorithms/Dichotomy.jl:65]: Also know as "Aneja and Nair method (1979)". Indeed, just named "dichotomy" is preferable.

Comment (odow) [reply src/algorithms/Dichotomy.jl:65]: Renamed here: https://github.com/jump-dev/MultiObjectiveAlgorithms.jl/pull/40

**Pull request closed without merging**